Public/Src/Engine/Scheduler/PipHistoricPerfData.cs

```diff
@@ -176,23 +177,30 @@ public PipHistoricPerfData MakeFresh()
         /// </summary>
         public PipHistoricPerfData Merge(PipHistoricPerfData other)
         {
-            return new PipHistoricPerfData(
-                DefaultTimeToLive,
-                GetMergeResult(DurationInMs, other.DurationInMs),
-                GetMergeResult(PeakMemoryInKB, other.PeakMemoryInKB),
-                (ushort)GetMergeResult(ProcessorsInPercents, other.ProcessorsInPercents),
-                GetMergeResult(DiskIOInKB, other.DiskIOInKB));
+            var durationResult = GetMergeResult(DurationInMs, other.DurationInMs);
+            var peakMemoryResult = GetMergeResult(PeakMemoryInKB, other.PeakMemoryInKB);
+            var processorInPercentResult = GetMergeResult(ProcessorsInPercents, other.ProcessorsInPercents);
+            var diskIOResult = GetMergeResult(DiskIOInKB, other.DiskIOInKB);
+
+            return new PipHistoricPerfData(DefaultTimeToLive, durationResult, peakMemoryResult, (ushort)processorInPercentResult, diskIOResult);
         }
 
         private static uint GetMergeResult(uint oldData, uint newData)
         {
-            // An asymmetric merge that goes up fast but decreases slowly.
-            if (newData > oldData)
+            try
             {
-                return (uint)(newData + (ulong) oldData) / 2;
-            }
+                // An asymmetric merge that goes up fast but decreases slowly.
+                if (newData > oldData)
+                {
+                    return (uint)(newData + (ulong) oldData) / 2;

```

Comment:
how about changing to `newData/2 + oldData/2`?

Reply:
Indeed.